README.md

```diff
@@ -1,2 +1,66 @@
 # triton-mysql
-MySQL designed for container-native deployment on Joyent's Triton platform
+
+MySQL designed for container-native deployment on Joyent's Triton platform.
+
+### Running a new cluster
+
+Starting a new cluster is easy. Just run `docker-compose up -d` and in a few moments you'll have a running MySQL primary. Both the primary and replicas are described as a single `docker-compose` service. During startup, [Containerbuddy](http://containerbuddy.io) will ask Consul if an existing primary has been created. If not, the node will initialize as a new primary and all future nodes will self-configure replication with the primary in their `onStart` handler.
+
+Run `docker-compose scale mysql=2` to add a replica (or more than one!). Replication in this architecture uses [Global Transaction Idenitifers (GTID)](https://dev.mysql.com/doc/refman/5.7/en/replication-gtids.html) rather than binlog positioning as this allows replicas to autoconfigure their position within the binlog. A primary that has the entire execution history can bootstrap a replica with no additional work. The replicas' `onChange` handler will automatically move replication to a new primary if one is created.
+
+So long as we're working with a new cluster that has never rotated the binlog on the primary (and where there's not too much data yet), the new replicas will be able to catch up to the primary without further intervention. But if you're bringing up a replica in an existing cluster, you'll need to try the following workflow.
+
+### Adding a replica to an existing cluster
+
+A primary that has rotated the binlog or simply has a large binlog will be impractical to use to bootstrap replication without copying data first. In this case we're going to [copy the MySQL data directory](https://dev.mysql.com/doc/refman/5.7/en/replication-gtids-failover.html) to the new replica's file system.
+
+In order to safely snapshot MySQL, we need to prevent new writes. In order to avoid downtime for the application, we recommend using one of the other replicas as a source for the data directory. The process that's been automated here is as follows:
+
+- Write a "lock" key to Consul which the replica will look for so that it doesn't complete replication setup until it has the data transfered from the source.
+- Start a new replica node. The new replica will see there's an existing primary, but will pause the setup process until the lock key has been removed from Consul.
+- Use `docker exec` to run `STOP SLAVE` on the source node.
```

Comment:
> Also note that we'll need to do more than STOP SLAVE. We have to stop the entire mysqld, I think.

Actually it's not `STOP SLAVE` at all. That should read set the source read-only via `SET @@global.read_only = ON;`. Which is enough to make it safe to copy the datadir. The replica isn't running at this point.